rfcs/text/i18n.md

```diff
@@ -0,0 +1,298 @@
+- Start Date: 2020-07-14
+
+# Summary
+
+Introduce first class support for internationalisation and localisation within `carbon-react`.
+
+# Basic example
+
+The proposal is to introduce a new component `I18nProvider` that allows our consumers to provide a 
+[`react-i18next`](https://github.com/i18next/react-i18next) compatible<sup>[1] [2]</sup> `t` function.
+
+We will need to check each usage of our current `i18n-js` implementation to ensure the:
+* key makes sense;
+* translations are provided in each of our supported languages; and
+* interpolation/plural/formatting works correctly.
+
+`react-i18next` also uses [`Suspense`](https://reactjs.org/docs/concurrent-mode-suspense.html) to allow for async translation loading. We don't need to support this within `carbon-react` to adopt this approach, but we will use this in the future when we add `loading` states to our components.
+
+We will deprecate our `I18n` component. This component consumes `i18n-js` directly and formats any Markdown content.
+There are better alternatives for Markdown such as (`react-markdown`)[https://github.com/rexxars/react-markdown] which avoids `dangerouslySetInnerHTML`.
+Our consumers can use `react-markdown` and `react-i18next` in tandem. Removing the `I18n` from carbon allows our consumers to patch any 
+Markdown parsing vulnerabilities without requiring a new `carbon-react` release.
+
+We will also deprecate all [`i18n` helpers](https://github.com/Sage/carbon/blob/a59afc80fd7ac954b58da5cd9ef278239b5756c0/src/utils/helpers/i18n/i18n.js). Continuing to support these is beyond the scope of a component library and is better handled by a dedicated library such as [`numeral.js`](http://numeraljs.com/).
```

Comment:
What's the plan for the ones we use internally right now?  Like `formatDecimal` in the Decimal Component

https://github.com/Sage/carbon/blob/c7a51a2e53542c31790966c68d7e4d381640e0dd/src/__experimental__/components/decimal/decimal.component.js#L229

Reply:
We'll also update our components to use a dedicated library, in this case numeral.js. When we do this we'll move this helper to a `__internal__` folder, or even inline it within the component itself.